src/components/core/geoChart/geoChartDataLayers.ts

```diff
@@ -1,21 +1,134 @@
 // (C) 2019-2020 GoodData Corporation
+import uniqBy = require("lodash/uniqBy");
 import { Execution } from "@gooddata/typings";
 import mapboxgl from "mapbox-gl";
+import { DEFAULT_COLORS, getLighterColor } from "../../visualizations/utils/color";
 import {
-    DEFAULT_PUSHPIN_COLOR_PALETTE,
+    DEFAULT_PUSHPIN_COLOR_SCALE,
     DEFAULT_PUSHPIN_COLOR_VALUE,
     DEFAULT_PUSHPIN_COLOR_VALUE_KEY,
     DEFAULT_PUSHPIN_OPTIONS,
+    DEFAULT_PUSHPIN_SEGMENT_BY_VALUE_KEY,
     DEFAULT_PUSHPIN_SIZE_SCALE,
     DEFAULT_PUSHPIN_SIZE_VALUE,
     DEFAULT_PUSHPIN_SIZE_VALUE_KEY,
     PUSHPIN_STYLE_CIRCLE,
     PUSHPIN_STYLE_CIRCLE_COLOR,
     PUSHPIN_STYLE_CIRCLE_SIZE,
+    PUSHPIN_STYLE_CIRCLE_STROKE_COLOR,
 } from "../../../constants/geoChart";
+import { getHeaderItemName } from "../../../helpers/executionResultHelper";
 import { stringToFloat } from "../../../helpers/utils";
 import { IGeoDataIndex } from "../../../interfaces/GeoChart";
 
+function createPushPinBorderOptions(
+    executionResult: Execution.IExecutionResult,
+    geoDataIndex: IGeoDataIndex,
+): mapboxgl.Expression | string {
+    const { segmentBy } = geoDataIndex;
+    const data = executionResult.data as Execution.DataValue[][];
+    const colorData = (data[geoDataIndex.color] || []).map(stringToFloat);
+
+    if (colorData.length === 0) {
+        return DEFAULT_COLORS[0];
+    }
+
+    if (segmentBy !== undefined) {
+        const {
+            headerItems: [, attributeHeaderItems],
```

Comment:
This could be an issue if there is no measure.
Please refer to the fix here https://github.com/gooddata/gooddata-react-components/pull/1318/files#diff-6b220fe05c6963feb3422220c77bcfa9R54